### PR TITLE
improved hypr_chan config options

### PR DESCRIPTION
**Pull request by DaniD3v**
#### Describe your PR, what does it fix/add?
removed disable_hypr_chan and replaced it with force_hypr_chan = 0.

This removes 1 unnecessary config variable and adds the ability to select which hypr_chan you want to enforce.

Ok ik this is kinda useless but I really wanted to be able to select my favorite hypr_chan.

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
Removed disable_hypr_chan (but force_hypr_chan=true still works)

I can also add some compatibility code so that force_hypr_chan is automatically set to 0 when disable_hypr_chan is true

#### Is it ready for merging, or does it need work?
Ready

### CODE REVIEW

**Review by vaxerski — APPROVED** [review]
lgtm, wiki mr pls

**Comment by DaniD3v** [comment]
added wiki entry
https://github.com/hyprwm/hyprland-wiki/pull/360

**Comment by vaxerski** [comment]
thats actually nicer

**Review by vaxerski — COMMENTED** [review]
just slap it with clang-format pls

File: src/render/OpenGL.cpp [hunk 1]
```diff
@@ -1762,34 +1762,37 @@ void CHyprOpenGLImpl::renderSplash(cairo_t* const CAIRO, cairo_surface_t* const
...
-        texPath += std::string(USEANIME ? (distribution2(engine) == 0 ? "anime_" : "anime2_") : "");
-
-    // check if wallpapers exist
+    if (*PFORCEWALLPAPER == -1) {
+        std::random_device              dev;
+        std::mt19937                    engine(dev());
+        std::uniform_int_distribution<> distribution(0, 2);
+        std::uniform_int_distribution<> distribution_anime(1, 2);
+
+        if (PFORCEHYPRCHAN) {
```
Comment (vaxerski) [inline src/render/OpenGL.cpp:1788]: style: no {} around short ifs

**Review by vaxerski — COMMENTED** [review]
other than that lgtm

**Comment by vaxerski** [comment]
but you forgot clang-tidy :P

File: src/render/OpenGL.cpp [hunk 2]
```diff
@@ -1762,34 +1762,35 @@ void CHyprOpenGLImpl::renderSplash(cairo_t* const CAIRO, cairo_surface_t* const
...
+        std::mt19937                    engine(dev());
+        std::uniform_int_distribution<> distribution(0, 2);
+        std::uniform_int_distribution<> distribution_anime(1, 2);
+
+        if (PFORCEHYPRCHAN)
+            texPath += prefixes[distribution_anime(engine)];
+        else
+            texPath += prefixes[distribution(engine)];
+    } else
+        texPath += prefixes[*PFORCEWALLPAPER];
```
Comment (vaxerski) [inline src/render/OpenGL.cpp:1793]: didn't notice this, but this needs a `std::clamp`, if someone puts in 27 it will crash

Comment (DaniD3v) [reply src/render/OpenGL.cpp:1793]: k sure I was already thinking about that.

Is there a way to do that when reading the config? I'd prefer to also give the user a warning if something went wrong. From the code I read it looks like you're only validating the types? Would be kinda stupid to change it for such a small feature tho.

Also sorry for the formatting :/ that was kinda retarded

Comment (vaxerski) [reply src/render/OpenGL.cpp:1793]: nope we have to clamp here

File: src/config/ConfigManager.cpp [hunk 3]
```diff
@@ -111,7 +111,7 @@ void CConfigManager::setDefaultVars() {
     configValues["misc:mouse_move_focuses_monitor"].intValue       = 1;
     configValues["misc:render_ahead_of_time"].intValue             = 0;
     configValues["misc:render_ahead_safezone"].intValue            = 1;
-    configValues["misc:cursor_zoom_factor"].floatValue             = 1.f;
+    configValues["misc:cursor_zoom_factor"].floatValue              = 1.f;
```
Comment (khaneliman) [inline src/config/ConfigManager.cpp:114]: unintended ?

Comment (vaxerski) [reply src/config/ConfigManager.cpp]: likely during my rebase

File: src/render/OpenGL.cpp [hunk 4]
```diff
@@ -1762,34 +1762,35 @@ void CHyprOpenGLImpl::renderSplash(cairo_t* const CAIRO, cairo_surface_t* const
...
-
-    std::random_device              dev;
-    std::mt19937                    engine(dev());
-    std::uniform_int_distribution<> distribution(0, 2);
-    std::uniform_int_distribution<> distribution2(0, 1);
-
-    const bool                      USEANIME = *PFORCEHYPRCHAN || distribution(engine) == 0; // 66% for anime
+    static auto* const PNOSPLASH       = &g_pConfigManager->getConfigValuePtr("misc:disable_splash_rendering")->intValue;
+    static auto* const PFORCEHYPRCHAN  = &g_pConfigManager->getConfigValuePtr("misc:force_hypr_chan")->intValue;
+    static long const  PFORCEWALLPAPER = std::clamp(*&g_pConfigManager->getConfigValuePtr("misc:force_default_wallpaper")->intValue, (int64_t)-1, (int64_t)2);
```
Comment (vaxerski) [inline src/render/OpenGL.cpp:1767]: we can't make this static.

Only the pointer can be static.
```
const long FORCEWALLPAPER = std::clamp(*PFORCEWALLPAPER, -1, 2);
```

File: src/render/OpenGL.cpp [hunk 5]
```diff
@@ -1762,34 +1762,35 @@ void CHyprOpenGLImpl::renderSplash(cairo_t* const CAIRO, cairo_surface_t* const
...
-
-    std::random_device              dev;
-    std::mt19937                    engine(dev());
-    std::uniform_int_distribution<> distribution(0, 2);
-    std::uniform_int_distribution<> distribution2(0, 1);
-
-    const bool                      USEANIME = *PFORCEHYPRCHAN || distribution(engine) == 0; // 66% for anime
+    static auto* const PNOSPLASH       = &g_pConfigManager->getConfigValuePtr("misc:disable_splash_rendering")->intValue;
+    static auto* const PFORCEHYPRCHAN  = &g_pConfigManager->getConfigValuePtr("misc:force_hypr_chan")->intValue;
+    const long         PFORCEWALLPAPER = std::clamp(*&g_pConfigManager->getConfigValuePtr("misc:force_default_wallpaper")->intValue, (int64_t)-1, (int64_t)2);
```
Comment (vaxerski) [inline src/render/OpenGL.cpp:1767]: I don't think you understand at all.

```
static auto* const PFORCEWALLPAPER = &g_pConfigManager->getConfigValuePtr("misc:force_default_wallpaper")->intValue;
const long WALLPAPER = std::clamp(*PFORCEWALLPAPER, -1LL, 2LL);
```

**Comment by vaxerski** [comment]
jesus christ I gave you code and you ignored it and now made invalid code

**Review by vaxerski — APPROVED** [review]

**Pull request merged**